--- readpass.c	2017-03-19 22:39:27.000000000 -0400
+++ readpass.c	2017-04-08 16:36:27.000000000 -0400
@@ -119,50 +119,18 @@
 read_passphrase(const char *prompt, int flags)
 {
 	char *askpass = NULL, *ret, buf[1024];
-	int rppflags, use_askpass = 0, ttyfd;
-
-	rppflags = (flags & RP_ECHO) ? RPP_ECHO_ON : RPP_ECHO_OFF;
-	if (flags & RP_USE_ASKPASS)
-		use_askpass = 1;
-	else if (flags & RP_ALLOW_STDIN) {
-		if (!isatty(STDIN_FILENO)) {
-			debug("read_passphrase: stdin is not a tty");
-			use_askpass = 1;
-		}
-	} else {
-		rppflags |= RPP_REQUIRE_TTY;
-		ttyfd = open(_PATH_TTY, O_RDWR);
-		if (ttyfd >= 0)
-			close(ttyfd);
-		else {
-			debug("read_passphrase: can't open %s: %s", _PATH_TTY,
-			    strerror(errno));
-			use_askpass = 1;
-		}
+	if (getenv(SSH_ASKPASS_ENV)) {
+		askpass = getenv(SSH_ASKPASS_ENV);
 	}
-
-	if ((flags & RP_USE_ASKPASS) && getenv("DISPLAY") == NULL)
-		return (flags & RP_ALLOW_EOF) ? NULL : xstrdup("");
-
-	if (use_askpass && getenv("DISPLAY")) {
-		if (getenv(SSH_ASKPASS_ENV))
-			askpass = getenv(SSH_ASKPASS_ENV);
-		else
-			askpass = _PATH_SSH_ASKPASS_DEFAULT;
-		if ((ret = ssh_askpass(askpass, prompt)) == NULL)
-			if (!(flags & RP_ALLOW_EOF))
-				return xstrdup("");
-		return ret;
+	else {
+		askpass = _PATH_SSH_ASKPASS_DEFAULT;
 	}
 
-	if (readpassphrase(prompt, buf, sizeof buf, rppflags) == NULL) {
-		if (flags & RP_ALLOW_EOF)
-			return NULL;
-		return xstrdup("");
+	if ((ret = ssh_askpass(askpass, prompt)) == NULL) {
+		if (!(flags & RP_ALLOW_EOF)) {
+			return xstrdup("");
+		}
 	}
-
-	ret = xstrdup(buf);
-	explicit_bzero(buf, sizeof(buf));
 	return ret;
 }
 
